specification/loadtestservice/resource-manager/Microsoft.LoadTestService/stable/2022-12-01/loadtestservice.json

```diff
@@ -2,985 +2,1075 @@
       "type": "object",
+      "description": "A collection of related endpoints from the same service for which the Batch service requires outbound access.",
       "properties": {
-        "properties": {
-          "description": "Check quota availability response properties.",
-          "$ref": "#/definitions/CheckQuotaAvailabilityResponseProperties",
-          "x-ms-client-flatten": true
-        }
-      },
-      "allOf": [
-        {
-          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/Resource"
+        "category": {
+          "type": "string",
+          "description": "The type of service that Azure Load Testing connects to.",
+          "readOnly": true
+        },
+        "endpoints": {
+          "type": "array",
+          "description": "The endpoints for this service to which the Batch service makes outbound calls.",
+          "items": {
+            "$ref": "#/definitions/EndpointDependency"
+          },
+          "readOnly": true,
+          "x-ms-identifiers": [
+            "domainName"
+          ]
         }
-      ]
+      }
     },
-    "CheckQuotaAvailabilityResponseProperties": {
-      "description": "Check quota availability response properties.",
+    "OutboundEnvironmentEndpointCollection": {
       "type": "object",
+      "description": "Values returned by the List operation.",
       "properties": {
-        "isAvailable": {
-          "description": "True/False indicating whether the quota request be granted based on availability.",
-          "type": "boolean"
+        "value": {
+          "type": "array",
+          "description": "The collection of outbound network dependency endpoints returned by the listing operation.",
+          "items": {
+            "$ref": "#/definitions/OutboundEnvironmentEndpoint"
+          },
+          "readOnly": true,
+          "x-ms-identifiers": []
         },
-        "availabilityStatus": {
-          "description": "Message indicating additional details to add to quota support request.",
-          "type": "string"
+        "nextLink": {
+          "type": "string",
+          "format": "uri",
+          "description": "The continuation token."
         }
       }
     },
     "QuotaBucketRequest": {
-      "description": "Request object of new quota for a quota bucket.",
       "type": "object",
+      "description": "Request object of new quota for a quota bucket.",
       "properties": {
         "properties": {
-          "description": "New quota request request properties.",
           "$ref": "#/definitions/QuotaBucketRequestProperties",
+          "description": "Request object of new quota for a quota bucket.",
           "x-ms-client-flatten": true
         }
       },
       "allOf": [
         {
-          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/Resource"
+          "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/Resource"
```

Comment:
Its weird that this is inheriting resource if its the respone only to POST requests. Did I understand the usage correctly? Can we do a breaking change to simplify it, if only preview versions were released so far?

Reply:
GA version is published already and there QuotaBucketRequest has a allof relation with Resource.

Reply:
From v3 -> v4/v5, only change is Resource.Id had an additional `"format": "arm-id"`, which has no impact on service. It does have implication on SDK where some of the language (C#) will type it to a `ResourceIdentifier` rather than string, though it does have override of implicit assignment operator. So technically not breaking.

Reply:
Removed inheritance relation with Resource in all request model.